metrics/options.go

```diff
@@ -58,8 +69,35 @@ func defaultOptions() *options {
 		errorHandler: func(err error) {
 			logger.WithError(err).Error(err.Error())
 		},
-		metricSampleRate: defaultMetricSampleRate,
+		sampleRate: defaultMetricSampleRate,
+	}
+}
+
+// parseMetricOptions processes the provided metric options and returns a configured metricOptions object.
+// This function combines globally defined defaults with any user-provided option overrides.
+// Notes:
+//   - If no options are provided, the returned metricOptions will use:
+//   - No tag
+//   - Global sample rate (opts.sampleRate) as the default sampling rate
+//   - Any provided options will override these defaults for the specific metric
+func parseMetricOptions(options ...MetricOptions) metricOptions {
```

Comment:
Rename function to stay consistent with general options handling

```suggestion
// resolveMetricOptions processes the provided metric options and returns a configured metricOptions object.
// This function combines globally defined defaults with any user-provided option overrides.
// Notes:
//   - If no options are provided, the returned metricOptions will use:
//   - No tag
//   - Global sample rate (opts.sampleRate) as the default sampling rate
//   - Any provided options will override these defaults for the specific metric
func resolveMetricOptions(options ...MetricOptions) metricOptions {
```